pkg/controlplane/peer/client.go

```diff
@@ -44,24 +45,60 @@ type RemoteServerAuthorizationResponse struct {
 	AccessToken string
 }
 
-// authorize a request for accessing a peer exported service, yielding an access token.
-func (c *Client) Authorize(req *api.AuthorizationRequest) (*RemoteServerAuthorizationResponse, error) {
-	body, err := json.Marshal(req)
-	if err != nil {
-		return nil, fmt.Errorf("unable to serialize authorization request: %w", err)
+func (c *Client) getResponse(
```

Comment:
Can you write a comment for the function,  what it returns in case of multiple gateways (in the function description or near the code)